org.eclipse.scanning.api/src/org/eclipse/scanning/api/database/Bean.java

```diff
@@ -0,0 +1,157 @@
+package org.eclipse.scanning.api.database;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * 
+ * This class links to a database bean using reflection.
+ * It means that an arbitrary bean may be used outside the {@link IExperimentDatabaseService}
+ * which is not in the classpath at the time it is used. The Bean instance
+ * is then transmitted to the IExperimentDatabaseService whose classloader is
+ * able to load the class referenced and set the fields using reflection.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+public class Bean {
+	
+	private String             beanClass;
+    private Map<String,Object> data;
+	
+	public Bean() {
+		this(null);
+	}
+	
+	public Bean(String beanClass) {
+		this.beanClass = beanClass;
+		this.data      = new HashMap<>();
+	}
+	
+	/**
+	 * Shortcut method used for when you already have an instance of the bean.
+	 * @param theBean
+	 * @throws NoSuchMethodException
+	 * @throws SecurityException
+	 * @throws IllegalAccessException
+	 * @throws IllegalArgumentException
+	 * @throws InvocationTargetException
+	 */
+	public Bean(Object theBean) throws NoSuchMethodException, 
+										SecurityException, 

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.SecurityException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)